core/src/main/java/tc/oc/pgm/fireworks/FireworkUtil.java

```diff
@@ -0,0 +1,47 @@
+package tc.oc.pgm.fireworks;
+
+import com.google.common.base.Preconditions;
+import javax.annotation.Nonnull;
+import org.bukkit.Bukkit;
+import org.bukkit.FireworkEffect;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Firework;
+import org.bukkit.inventory.meta.FireworkMeta;
+
+public class FireworkUtil {
+
+  public static @Nonnull Firework spawnFirework(
+      @Nonnull Location location, @Nonnull FireworkEffect effect, int power) {
+    Preconditions.checkNotNull(location, "location");
+    Preconditions.checkNotNull(effect, "firework effect");
+    Preconditions.checkArgument(power >= 0, "power must be positive");
+
+    FireworkMeta meta = (FireworkMeta) Bukkit.getItemFactory().getItemMeta(Material.FIREWORK);
+    meta.setPower(power);
+    meta.addEffect(effect);
+
+    Firework firework = (Firework) location.getWorld().spawnEntity(location, EntityType.FIREWORK);
+    firework.setFireworkMeta(meta);
+
+    return firework;
+  }
+
+  public static @Nonnull Location getOpenSpaceAbove(@Nonnull Location location) {
```

Comment:
Don't we have some `Locations` or `LocationUtil` class?

Reply:
@Electroid I was unable to find a `Locations` or `LocationUtil` class. Perhaps you have some guidance on this? I placed both of the FireworkUtil methods into the `FireworkMatchModule` for now.